File: package/main/src/IP/getIpClass.ts
/**
 * Gets the IP address class (A, B, C, D, or E)
 * @param {string} ip - IPv4 address
 * @returns {string} IP class ('A', 'B', 'C', 'D', 'E', or empty string for invalid IP)
 */
export const getIpClass = (ip: string): string => {
  if (!ip) {
    return "";
  }

  // Validate IP format
  const parts = ip.split(".");
  if (parts.length !== 4) {
    return "";
  }

  const firstOctet = Number.parseInt(parts[0]);
  if (Number.isNaN(firstOctet) || firstOctet < 0 || firstOctet > 255) {
    return "";
  }

  // Check each class range
  switch (true) {
    case firstOctet === 0: {
      return "";
    } // Reserved
    case firstOctet < 128: {
      return "A";
    } // 1-127
    case firstOctet < 192: {
      return "B";
    } // 128-191
    case firstOctet < 224: {
      return "C";
    } // 192-223
    case firstOctet < 240: {
      return "D";
    } // 224-239
    default: {
      return "E";
    } // 240-255
  }
};

File: package/umt_python/src/ip/get_ip_class.py
def get_ip_class(ip: str) -> str:
    """
    Gets the IP address class (A, B, C, D, or E).

    Args:
        ip: IPv4 address

    Returns:
        IP class ('A', 'B', 'C', 'D', 'E', or empty string for invalid IP)

    Example:
        >>> get_ip_class("10.0.0.1")
        'A'
        >>> get_ip_class("172.16.0.1")
        'B'
        >>> get_ip_class("192.168.1.1")
        'C'
    """
    if not ip:
        return ""

    parts = ip.split(".")
    if len(parts) != 4:
        return ""

    try:
        first_octet = int(parts[0])
    except ValueError:
        return ""

    if first_octet < 0 or first_octet > 255:
        return ""

    if first_octet == 0:
        return ""
    if first_octet < 128:
        return "A"
    if first_octet < 192:
        return "B"
    if first_octet < 224:
        return "C"
    if first_octet < 240:
        return "D"
    return "E"
